From 5a1ee5466a22b36a784d00f68a4d9d49f0605e96 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Mon, 2 Mar 2026 02:33:56 +0800
Subject: [PATCH] Disable AV1 decoder on A16 QPR2 temporarily

As it does not work properly when playing 1080p+ videos (at least on ARL)

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 c2_store/src/mfx_c2_store.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/c2_store/src/mfx_c2_store.cpp b/c2_store/src/mfx_c2_store.cpp
index a8352ba..731f15e 100644
--- a/c2_store/src/mfx_c2_store.cpp
+++ b/c2_store/src/mfx_c2_store.cpp
@@ -24,6 +24,7 @@
 #include "mfx_debug.h"
 #include "mfx_c2_debug.h"
 #include "mfx_c2_component.h"
+#include <android/api-level.h>
 #include <android-base/properties.h>
 #include <cutils/properties.h>
 
@@ -269,6 +270,9 @@ static bool isCodecSupportedByGPU(const std::string codec) {
     // Every modern Intel GPU supports MPEG2
     if (codec == "c2.intel.mp2.decoder") return true;
 
+    // Disable AV1 decoder on A16, as it does not work properly when playing 1080p+ videos (at least on ARL)
+    if (codec == "c2.intel.av1.decoder" && android_get_device_api_level() >= 36) return false;
+
     const std::string supported_codecs = ::android::base::GetProperty("ro.waydroid.hwcodecs", "");
     const bool is_encoder = (codec.find(".encoder") != std::string::npos);
 
-- 
2.53.0

